### ISSUE TITLE
Keras Model giving half results

### ISSUE THREAD

**Issue by Subhambh** [tensorflow/tfjs#455]
Browesr version : 67.0.3396.87

I am trying to do image segmentation of cell images to identify cells in tensorflow.js using data provided in https://www.kaggle.com/c/data-science-bowl-2018/data i have build a keras model consisting of unet architecture my model is able to do image segmentation of only upper half of the image and rest half it is showing entirely masked or unmasked based on my threshold. The result is same for all the images i have tried.
![screenshot 7](https://user-images.githubusercontent.com/40203444/41638563-ac51b6c0-7477-11e8-8472-c5b3a349dbe1.png)

I know my model is correct because once (ONLY) during testing i got entirely masked image using the above image in the browser.
Please help.
code in js:

<html>
	<title> TensorFlow </title>
	<head>
	
	<script src="https://cdn.jsdelivr.net/npm/@tensorflow/tfjs@0.11.6"> </script>
	<img id='Cells' src='../static/sample.PNG' alt='BVI Cells' width="128" height="128"> 
	<canvas id="myCanvas" width="128" height="128" ></canvas> 

	<script async>
		var cells = document.getElementById('Cells',1);
		console.log(cells.alt);
		
		var img = tf.fromPixels(cells);
		img = img.reshape([1,128,128,3]);
		img = tf.cast(img, 'float32');
		
		async function predict(image) {
			var c = document.getElementById("myCanvas");
			var ctx = c.getContext("2d");
			var r,g,b;
			
			var exam = image;
			model = await tf.loadModel ("{{ url_for('static', filename='model.json') }}");
			var result = model.predict(exam);
			const predictions = Array.from(result.dataSync());
			
			
			var index = 0;
			var x = new Array(128);
			for (var i = 0; i < 128; i++) {
				x[i] = new Array(128);
				for (var j = 0; j < 128; j++) {
					if (predictions[index] > 0.4) {
						x[i][j] = predictions[index]; 
						r = 255;
						g = 255;	
						b = 255;	
						ctx.fillStyle = "rgba("+r+","+g+","+b+", 1)"; 
						ctx.fillRect( j, i, 1, 1 );
					} else {
						x[i][j] = 0;
						r = 0;
						g = 0;	
						b = 0;	
						ctx.fillStyle = "rgba("+r+","+g+","+b+", 1)"; 
						ctx.fillRect( j, i, 1, 1 );
					}
					index = index+1;
				}
			}
			console.log(x);
			console.log(result);
		    return result;
		}
		
		
	    predict(img);
		
		
		
    </script>
	</head>
	<body>
	</body>
</html>

**Comment by pyu10055** [comment]
Based on the code, it is possible the prediction for the second half of the image is relatively high > 0.4, which cause all pixels to be white.

Can you print the predictions out? also it is possible that you might need to normalize the image data depends on the model.

**Comment by caisq** [comment]
+1 to what @pyu10055 said. What are the shapes of `exam` and `result`?

**Comment by nsthorat** [comment]
Hi @Subhambh, just wanted to ping this again before closing it out.

**Comment by tafsiri** [comment]
Closing this for inactivity. @Subhambh feel free to reopen if needed

**Comment by karelp** [comment]
Hi @pyu10055 @tafsiri , I've encountered this issue as well with tfjs 0.14.1. The problem seem to be pooling layers and/or strides on the webgl backend. As soon as I use a strided convolution or max pooling in my Keras model, the output of the pooling layer gets cut in half just as shown in the screenshots of OP. The result of the bottom half has a uniform value for each channel, however the concrete number seems to be pretty random. Note that using `tf.setBackend('cpu')` fixes the issue, the problem seems to be solely in the webgl backend. I can prepare a test project that manifests the issue if needed.

Excerpt from my model.summary():
```
Layer (type)                    Output Shape         Param #     Connected to                     
==================================================================================================
input (InputLayer)              (None, 256, 256, 3)  0                                            
__________________________________________________________________________________________________
conv2d_1 (Conv2D)               (None, 256, 256, 16) 448         input[0][0]                      
__________________________________________________________________________________________________
leaky_re_lu_1 (LeakyReLU)       (None, 256, 256, 16) 0           conv2d_1[0][0]                   
__________________________________________________________________________________________________
resBlock_0_downsample_conv1 (Conv2D) (None, 256, 256, 16) 272    leaky_re_lu_1[0][0]              
__________________________________________________________________________________________________
leaky_re_lu_2 (LeakyReLU)       (None, 256, 256, 16) 0           resBlock_0_downsample_conv1[0][0] <--- output OK
__________________________________________________________________________________________________
resBlock_0_downsample_pool1 (MaxPooling2D) (None, 128, 128, 16) 0           leaky_re_lu_2[0][0]    <--- output corrupted              
__________________________________________________________________________________________________
.
.
.
```
